tests/integration/test_jenkins.py

```diff
@@ -98,3 +99,23 @@ async def test_jenkins_automatic_update(
 
     updated_workload_version = updated_app_status.workload_version
     assert updated_workload_version == latest_jenkins_lts_version, "The Jenkins should be updated."
+
+
+async def test_rotate_password_action(unit_web_client: UnitWebClient):
+    """
+    arrange: given a jenkins API session that is connected.
+    act: when rotate password action is called.
+    assert: the session is invalidated and new password is returned.
+    """
+    session = unit_web_client.client.requester.session
+    session.get(f"{unit_web_client.web}/manage/")
+    action: Action = await unit_web_client.unit.run_action("rotate-credentials")
+    await action.wait()
+    new_password: str = action.results["password"]
+
+    assert unit_web_client.client.password != new_password, "Password not rotated"
+    result = session.get(f"{unit_web_client.web}/manage/")
+    assert result.status_code == 403, "Session not cleared"
+    new_client = jenkinsapi.jenkins.Jenkins(unit_web_client.web, "admin", new_password)
+    result = new_client.requester.get_url(f"{unit_web_client.web}/manage/")
+    assert result.status_code == 200, "Invalid password"
```

Comment:
Should whether the new password is working be tested?

Reply:
The line `new_client = jenkinsapi.jenkins.Jenkins(unit_web_client.web, "admin", new_password)` will fail to instantiate when the new password doesn't work :)